### DISCUSSION TITLE
Loading failed in next.js app in Safari browser

### DISCUSSION THREAD

**Discussion by Seedovan19** [supabase/supabase#14017] [Questions]
I followed this tutorial to get images from supabase bucket: https://supabase.com/docs/guides/getting-started/quickstarts/nextjs

Here is my next js code:
```
import { createClient } from "@supabase/supabase-js";

const supabase = createClient(
  "project url",
  "project api key (anon public)",
);

export async function getServerSideProps() {
  const { data } = await supabase.from("images").select();
  console.log(data);
  return {
    props: {
      pictures_test: data,
    },
  };
}

```
The problem is I get **null** values in my data variable.

I also tried to upload image to the bucket but I get this errors in Safari: 

`
[Error] Unhandled Promise Rejection: TypeError: Load failed
`

`
[Error] Fetch API cannot load https://my-url/storage/v1/object/images/2 due to access control checks.
`

I tried to debug it with Postman but I got this error there:

```
{
    "code": "42P01",
    "details": null,
    "hint": null,
    "message": "relation \"public.images\" does not exist"
}
```
Although I have a public bucket called images inside my project. I selected policies to give anon users all the access to bucket.
What can I do to solve the issue?

**Comment by GaryAustin1** [comment]
In your first code you are loading from a table in public schema called images.   This call has nothing to do with storage or buckets.  You need to access it with the storage API calls.

When you test with Postman you need to provide the authorization and apikey headers.  It would help if you showed the call details.

**Reply by Seedovan19** [reply]
Thanks for the reply, got it. I changed the code to this, now the data variable is not null, but returns an empty list (I have got one image in my images bucket):

```
export async function getServerSideProps() {
  const { data } = await supabase.storage.from("images").list("", {
    limit: 100,
  });

  return {
    props: {
      picturesTest: data,
    },
  };
}
```

I provided the api key and authorization header in my postman request. The url was wrong too (that's why I got "relation \"public.images\" does not exist"), I did not use the storage API calls. What is the right call to list all the images?

I tried this request with the "Authorization" header (Authorization: Bearer [api-token]) in my Postman but I get message: "The resource was not found":

`https://project-url/storage/v1/object/list/images?apikey=project-api-key`
